xrpl/models/amounts/amount.py

```diff
@@ -4,15 +4,17 @@
 counterparty.
 """
 
-from typing import Union, cast
+from typing import Union
+
+from typing_extensions import TypeGuard
 
 from xrpl.models.amounts.issued_currency_amount import IssuedCurrencyAmount
 from xrpl.models.amounts.mpt_amount import MPTAmount
 
 Amount = Union[IssuedCurrencyAmount, MPTAmount, str]
 
 
-def is_xrp(amount: Amount) -> bool:
+def is_xrp(amount: Amount) -> TypeGuard[str]:
```

Comment:
Have you found TypeGuard useful? I'm not convinced that it improves readability in this snippet.

I get the benefits with composite type examples listed here: https://typing.python.org/en/latest/spec/narrowing.html#typeguard, but I feel these types are simple enough to read.

Reply:
All the `# type: ignore`s I was able to remove were because I switched these to typeguards. It's not about readability, it's about `mypy` being able to understand the typing better if these functions are used.